Doc: Remove title text from internal links

rffontenelle opened the pull request:
Rely on the title of the linked internal page instead of putting the title. Sphinx will render with the title correctly, and this will reduce work for translators

Please backport it to 3.11 and 3.10 as well.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNN: Summary of the changes made
```

Where: gh-NNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->

CAM-Gerlach commented on Doc/whatsnew/3.10.rst:
```diff
@@ -2152,8 +2152,7 @@ Porting to Python 3.10
   ``#``: ``es#``, ``et#``, ``s#``, ``u#``, ``y#``, ``z#``, ``U#`` and ``Z#``.
-  See :ref:`Parsing arguments and building values
-  <arg-parsing>` and the :pep:`353`.
+  See :ref:`arg-parsing` and the :pep:`353`.
```
```suggestion
  See :ref:`arg-parsing` and :pep:`353`.
```

If we're changing this line anyway, the definite article is better left out here

CAM-Gerlach commented on Doc/whatsnew/3.10.rst:
```diff
@@ -2199,7 +2197,7 @@ Porting to Python 3.10
   ``pyfpe.h``, and ``pytime.h`` have been moved to the ``Include/cpython``
   directory. These files must not be included directly, as they are already
-  included in ``Python.h``: :ref:`Include Files <api-includes>`. If they have
+  included in ``Python.h``: :ref:`api-includes`. If they have
```
```suggestion
  included in ``Python.h``; see :ref:`api-includes`. If they have
```

The current syntax doesn't really make grammatical sense, so I suggest fixing it.

CAM-Gerlach reviewed (commented):
LGTM; two suggested fixes on the lines being changed.

rffontenelle committed:
Doc: Remove title text from internal links

Rely on the title of the linked internal page instead of putting the title. Sphinx will render with the title correctly, and this will reduce work for translators

CAM-Gerlach reviewed (approved):
LGTM, thanks @rffontenelle
